### ISSUE TITLE
Unable to fetch user & groups

### ISSUE THREAD

**Issue by akkitech** [jsumners/node-activedirectory#56]
I am using node js activedirectory module for AD authentication and fetching users & groups. Able to authenticate successfully but unable to get users & groups using .findUser(), .findUsers(), .findGroups() methods. I've added few users and a group using ADSI editor in windows 10. Any clue what could be going wrong?

Screenshot of ADSI Editor :

![image](https://user-images.githubusercontent.com/11798883/69871379-a513de00-12d8-11ea-8233-fcd776e45e60.png)

Code :

```
var ActiveDirectory = require('activedirectory');
var config = {
url: 'ldap://localhost',
baseDN: 'cn=admin,dc=ec2,dc=internal',
username: 'cn=akshay,cn=testContainer,cn=admin,dc=ec2,dc=internal',
password: '123456'
};

var ad = new ActiveDirectory(config);

var username = 'cn=akshay,cn=testContainer,cn=admin,dc=ec2,dc=internal';
var password = '123456';

ad.authenticate(username, password, function (err, auth) {
if (err) {
console.log('ERROR: ' + JSON.stringify(err));
return;
}

if (auth) {
    console.log('Authenticated!');
    ad.findUser({
        dn: "cn=akshay,cn=testContainer,cn=admin,dc=ec2,dc=internal"
    }, function (err, user) {
        if (err) {
            console.log('ERROR: ' + JSON.stringify(err));
            return;
        }

        if (!user) console.log('User: not found.');
        else console.log(JSON.stringify(user));
    });

} else {
    console.log('Authentication failed!');
}
});
```

Output :

Authenticated!
User: not found.

**Comment by jsumners** [comment]
Have you tried setting a search base and/or scope?

**Comment by akkitech** [comment]
Yes, passed baseDN as cn=testContainer,cn=admin,dc=ec2,dc=internal still no results.

**Comment by jurjendijkstra** [comment]
Perhaps a dumb answer, but I wonder if things are case sensitive and would try, just for curiosity:
"cn=TestContainer,cn=ADMIN,dc=EC2,dc=INTERNAL" same case as in the ADSI editor screenshot

**Comment by akkitech** [comment]
Thank you @jurjendijkstra , Already tried with upper case but no luck.

**Comment by jurjendijkstra** [comment]
Hmm sorry.  Another guess: you are passing an object { dn: "..."} to findUser but according to the doc you should pass a string, and the value of that string may be either the sAMAccountName, userPrincipalName  or dn.  Like this: 
 ad.findUser("cn=akshay,cn=testContainer,cn=admin,dc=ec2,dc=internal", callback)

**Comment by akkitech** [comment]
Thank you @jurjendijkstra , Tried with a DN string still no luck. 

Can you help me in finding my sAMAccountName / userPrincipalName in ADSI editor? I am unable to locate this option.

**Comment by akkitech** [comment]
OR sAMAccountName is just the **name** that was used while adding a new user in ADSI? i.e akshay in this case?
